jwst/photom/photom.py

```diff
@@ -497,6 +503,17 @@ def calc_miri(self, ftab):
             # Update the science dq
             self.input.dq = np.bitwise_or(self.input.dq, ftab.dq)
 
+            # Check if reference file contains time dependent correction
+            if np.any(ftab.timecoeff_ch1['binwave']) and self.mrs_time_correction:
+                log.info("Applying MRS IFU time dependent correction.")
+                mid_time = self.input.meta.exposure.mid_time
+                correction = ifu_mrs.time_correction(self.input, self.detector,
+                                                     ftab, mid_time)
+                self.input.data /= correction
```

Comment:
Just so I understand the overall approach here, is the regular flux calibration correction applied first, and then the time-dependent correction is applied separately here as an additional correction?

Reply:
yes.  Normal photom correction and if the time coefficients for band are present then make the time_correction. Unless the user set the mrs_time_correction to False